src/mono/System.Private.CoreLib/src/System/Type.Mono.cs

```diff
@@ -134,8 +134,22 @@ internal virtual FieldInfo GetField(FieldInfo fromNoninstanciated)
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         private static extern Type internal_from_handle(IntPtr handle);
 
-        [Intrinsic]
-        public static bool operator ==(Type? left, Type? right) => left == right;
+        public static bool operator ==(Type? left, Type? right)
```

Comment:
We could move this to the common Type.cs under `#if !CORECLR`. This is the same implementation that NativeAOT uses:

https://github.com/dotnet/runtimelab/blob/22dab87e510f56ffc53ff2b8d72b995c9127bf60/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Type.CoreRT.cs#L86-L102

(Note we want to use `.IsRuntimeImplemented()` since we don't test for `RuntimeType` in the shared codebase, and keep `[Intrinsic]` because this is trivial for codegen to optimize into reference equality if one of the types comes out of `Object.GetType` or `Type.GetTypeFromHandle`).

Reply:
Thanks, I was not aware of that.   I've now updated the patch to use the NativeAOT implementation in the common Type.cs under #if !CORECLR as you suggested.  This still fixes the problems I was seeing.